server/controllers/finance/debtors.js

```diff
@@ -152,13 +175,106 @@ function fetchInvoices (req, res, next){
       res.status(200).json(rows);
     })
     .catch(next)
-    .done();             
+    .done();
+}
+
+/**
+* Fetch Debtor Group Invoices
+*
+* @function fetchGroupInvoices
+*
+* @desc This function is responsible for getting all invoices of a specified debtor group
+*/
+function fetchGroupInvoices(req, res, next) {
+  getGroupInvoices(req.params.id)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(function (err) {
+    next(err);
+  })
+  .done();
+}
+
+/**
+* Get Debtor group invoices
+*/
+function getGroupInvoices (id) {
+  var def = q.defer();
```

Comment:
You can write this function without creating a new `defer()` object.  It would look like this (pseudocode)
```js
function getGroupInvoices(id) {

if (!id) { return q.reject(new Error('ERR_MISSING_REQUIRED_PARAMETERS')); }

// ... code ...

return db.exec(sql, [params])
 //... promise chain ....
};
```

Why is this better?
 1. Code clarity (if you know that [q.reject()](https://github.com/kriskowal/q/wiki/API-Reference#qrejectreason) and [q(https://github.com/kriskowal/q/wiki/API-Reference#q-object)]() can immediate create rejected/resolved promises).
 2. Error handling can be completely moved to outside this module.
 3. No dangling errors!